analyzer/decrypt_crypto.cc

```diff
@@ -234,59 +232,53 @@ std::vector<uint8_t> decrypt(std::vector<uint8_t> client_key, std::vector<uint8_
 	EVP_CipherInit_ex(ctx, NULL, NULL, client_key.data(), decryptInfo.nonce.data(), 0);
 
 	// Set the tag to be validated after decryption
-	EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_CCM_SET_TAG, tag_to_check.size(), tag_to_check.data());
+	EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_CCM_SET_TAG, tag_to_check_length, (void*)tag_to_check);
```

Comment:
Let's spell out the const cast,

```suggestion
	EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_CCM_SET_TAG, tag_to_check_length, const_cast<void*>(tag_to_check));
```

This requires that you change the type of `tag_to_check` to `const void*` in its declaration.